### PR TITLE
Patch out a method to bypass xenobio progression

### PR DESCRIPTION
**Pull request by RandomGamer123**
## About The Pull Request
Makes the slime transformation disease result in a grey slime instead of a random-coloured one, unless the infected person is a sentient non-monkey human

## Why It's Good For The Game
Using smoke to spread the advanced mutation toxin by the black slime allows distributing it to a large amount of monkeys at once, which then turn into random coloured slimes, bypassing xenobio progression, also see https://github.com/BeeStation/BeeStation-Hornet/pull/6299

In addition, given the description on the wiki https://tgstation13.org/wiki/Guide_to_xenobiology#Tier_5, having it be grey slimes is actually intended behavior as well

## Changelog

:cl:
fix: Removes a method that allows xenobiologists to produce slimes of every colour and bypassing the intended progression sequence by making the disease induced by advanced mutation toxin only produce grey slimes unless the infected person is a sentient non-monkey human
/:cl:

### CODE REVIEW

**Comment by RandomGamer123** [comment]
Hmm actually yes that sounds neat

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by TheVekter** [comment]
I think someone said a while back this wasn't going to be fixed since the xenobio overhaul is coming, but I'm not sure what exactly the plan is with that yet.

File: code/datums/diseases/transformation.dm [hunk 1]
```diff
@@ -243,9 +243,15 @@
 		if(3)
 			if(ishuman(affected_mob))
 				var/mob/living/carbon/human/human = affected_mob
-				if(human.dna.species.id != SPECIES_SLIMEPERSON && affected_mob.dna.species.id != SPECIES_STARGAZER && affected_mob.dna.species.id != SPECIES_LUMINESCENT)
+				if(human.dna.species.id != SPECIES_MONKEY && human.dna.species.id != SPECIES_SLIMEPERSON && affected_mob.dna.species.id != SPECIES_STARGAZER && affected_mob.dna.species.id != SPECIES_LUMINESCENT)
```
Comment (ShizCalev) [inline code/datums/diseases/transformation.dm:246]: ismonkey isslimeperson

Comment (RandomGamer123) [reply code/datums/diseases/transformation.dm]: isjellyperson actually to include stargazers and luminescents but :+1:

File: code/datums/diseases/transformation.dm [hunk 2]
```diff
@@ -243,9 +243,15 @@
 		if(3)
```
Comment (ShizCalev) [inline code/datums/diseases/transformation.dm:243]: Mind hitting the check above this to use the helper as well?

**Pull request merged**